File: Java Fundamentals Homeworks/05_Java_OOP/1.Geometry/PlaneShapes/Circle.java
package _1_Geometry.PlaneShapes;

import _1_Geometry.Vertices.Vertex2D;

public class Circle extends PlaneShape{

    private double radius;

    public Circle(Vertex2D vertex2D, double radius){
        super(vertex2D);
        setRadius(radius);
        this.perimeter = calculatingPerimeter();
        this.area = calculatingArea();
    }

    @Override
    protected double calculatingArea(){
        return Math.PI*this.radius*this.radius;
    }
    @Override
    protected double calculatingPerimeter() {
        return 2 * Math.PI * this.radius;
    }

    public double getRadius() {
        return radius;
    }

    public void setRadius(double radius) {
        this.radius = radius;
    }

}

File: 03_JS_Core/02_js_advanced/10_Classes_And_Members/circle.js
/**
 * Created by ivans on 02-Nov-16.
 */
class Circle {
    constructor(radius){
        this.radius = radius
    }
    get diameter() {
        return this.radius*2
    }
    set diameter(value){
        this.radius = value/2
    }
    get area(){
        return Math.PI * this.radius * this.radius
    }
}